File: frontend/src/types/diff.ts
export type DiffSummary = {
  changed: boolean
  bin_changed: boolean
  lib_changed: boolean
}

export type DiffTask = {
  task_id: string
  status: string
  summary: DiffSummary | null
  result_file_id: string | null
  result_text: string | null
  error: string | null
  created_at: string
  updated_at: string
}

export type CreateDiffTaskPayload = {
  left_file_id: string
  right_file_id: string
}

File: backend/app/schemas/diff.py
from datetime import datetime

from pydantic import BaseModel


class DiffTaskCreate(BaseModel):
    left_file_id: str
    right_file_id: str


class DiffSummary(BaseModel):
    changed: bool
    bin_changed: bool
    lib_changed: bool


class DiffTaskRead(BaseModel):
    task_id: str
    status: str
    summary: DiffSummary | None = None
    result_file_id: str | None = None
    result_text: str | None = None
    error: str | None = None
    created_at: datetime
    updated_at: datetime

